src/python/foglamp/device/http_south_device.py

```diff
@@ -0,0 +1,151 @@
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_LOGGER = logger.setup(__name__)
+
+_CONFIG_CATEGORY_NAME = 'http_south'
+_CONFIG_CATEGORY_DESCRIPTION = 'South Plugin HTTP Listener'
+_DEFAULT_CONFIG = {
+    'plugin': {
+         'description': 'http_south_device',
+         'type': 'string',
+         'default': 'http_south'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '6683',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': '0.0.0.0',
+    }
+}
+
+
+def plugin_info():
+    return {'name': 'http_south', 'version': '1.0', 'mode': 'async', 'type': 'device',
+            'interface': '1.0', 'config': _DEFAULT_CONFIG}
+
+
+def plugin_init(config):
+    """Registers HTTP Listener handler to accept sensor readings"""
+
+    _LOGGER.info("Retrieve HTTP Listener Configuration %s", config)
+
+    uri = config['uri']['value']
+    port = config['port']['value']
+
+    loop = asyncio.get_event_loop()
+
+    app = web.Application(middlewares=[middleware.error_middleware])
+    app.router.add_route('POST', '/', HttpSouthIngest.render_post)
+    handler = app.make_handler()
+    coro = loop.create_server(handler, uri, port)
+
+    return coro, handler
+
+
+def plugin_run(data):
+    pass
+
+def plugin_reconfigure(config):
+    pass
+
+
+def plugin_shutdown(data):
+    pass
+
+
+class HttpSouthIngest():
+    """Handles incoming sensor readings from HTTP Listner"""
+
+    @staticmethod
+    async def render_post(request):
+        """Store sensor readings from CoAP to FogLAMP
+
+        Args:
+            request:
+                The payload is a cbor-encoded array that decodes to JSON
+                similar to the following:
+
+                .. code-block:: python
+
+                    {
+                        "timestamp": "2017-01-02T01:02:03.23232Z-05:00",
+                        "asset": "pump1",
+                        "key": "80a43623-ebe5-40d6-8d80-3f892da9b3b4",
+                        "readings": {
+                            "velocity": "500",
+                            "temperature": {
+                                "value": "32",
+                                "unit": "kelvin"
+                            }
+                        }
+                    }
+        Example: curl -X POST http://localhost:6683 -d '{"timestamp": "2017-01-02T01:02:03.23232Z-05:00", "asset": "pump1", "key": "80a43623-ebe5-40d6-8d80-3f892da9b3b4", "readings": {"velocity": "500", "temperature": {"value": "32", "unit": "kelvin"}}}'
+        """
+        # TODO: The payload is documented at
+        # https://docs.google.com/document/d/1rJXlOqCGomPKEKx2ReoofZTXQt9dtDiW_BHU7FYsj-k/edit#
+        # and will be moved to a .rst file
+
+        code = web.HTTPInternalServerError.status_code
+        increment_discarded_counter = True
+        message = ''
+
```

Comment:
Be positive :)
`code = web.HTTPOk.status_code`
`message = {'result': 'success'}`

`web.HTTPInternalServerError.status_code` should be assigned in `except Exception as e:`